hw/misc/hermes.c

```diff
@@ -504,6 +618,13 @@ static void hermes_bar0_write(void *opaque, hwaddr addr, uint64_t val,
             return;
         }
 
+        if (val == HERMES_CMD_START) {
+            bar0_init.cmdctrl[engine].ehcmddone = HERMES_CMD_NOT_FINISHED;
+            qemu_mutex_lock(&hermes->ubpf[engine].bpf_mutex);
+            qemu_cond_signal(&hermes->ubpf[engine].bpf_cond);
+            qemu_mutex_unlock(&hermes->ubpf[engine].bpf_mutex);
+            bar0_init.cmdctrl[engine].ehcmdexec = HERMES_CMD_STOP;
```

Comment:
That should be set only when we finish the command (i.e., on `mark_exec_finished()`). Remember that the program is actually executed on a different thread, we don't want to set this register too early ;)

Reply:
In mark_exec_finished it is set to FINISHED. Maybe we can reduce it to just START and STOP?

Reply:
That would not be enough, as we can't detect a program in a running state.